lib/tasks/import.rake

```diff
@@ -0,0 +1,84 @@
+require 'csv'
+desc "Import customer from csv file"
+  task :import => [:environment] do
+   file = "db/data/customers.csv"
+    CSV.foreach(file, headers: true) do |row|
+      customer_hash = row.to_hash
+      customer = Customer.where(id: customer_hash["id"])
+      if customer.count == 1
+        customer.first.update_attributes(customer_hash)
+      else
+        Customer.create!(customer_hash)
+      end
+  end
+end
+
+desc "Import merchant from csv file"
+  task :import => [:environment] do
+   file = "db/data/merchants.csv"
+    CSV.foreach(file, headers: true) do |row|
+      merchant_hash = row.to_hash
+      merchant = Merchant.where(id: merchant_hash["id"])
+      if merchant.count == 1
+        merchant.first.update_attributes(merchant_hash)
+      else
+        Merchant.create!(merchant_hash)
+      end
+  end
+end
+
+desc "Import invoice from csv file"
+  task :import => [:environment] do
+   file = "db/data/invoices.csv"
+    CSV.foreach(file, headers: true) do |row|
+      invoice_hash = row.to_hash
+      invoice = Invoice.where(id: invoice_hash["id"])
+      if invoice.count == 1
+        invoice.first.update_attributes(invoice_hash)
+      else
+        Invoice.create!(invoice_hash)
+      end
+  end
+end
+
+desc "Import item from csv file"
+  task :import => [:environment] do
+   file = "db/data/items.csv"
+    CSV.foreach(file, headers: true) do |row|
+      item_hash = row.to_hash
+      item = Item.where(id: item_hash["id"])
+      if item.count == 1
+        item.first.update_attributes(item_hash)
+      else
+        Item.create!(item_hash)
+      end
+  end
+end
+
+desc "Import transaction from csv file"
+  task :import => [:environment] do
+   file = "db/data/transactions.csv"
+    CSV.foreach(file, headers: true) do |row|
+      transaction_hash = row.to_hash
+      transaction = Transaction.where(id: transaction_hash["id"])
+      if transaction.count == 1
+        transaction.first.update_attributes(transaction_hash)
+      else
+        Transaction.create!(transaction_hash)
+      end
+  end
+end
+
+desc "Import invoice_item from csv file"
+  task :import => [:environment] do
+   file = "db/data/invoice_items.csv"
+    CSV.foreach(file, headers: true) do |row|
+      invoice_item_hash = row.to_hash
+      invoice_item = InvoiceItem.where(id: invoice_item_hash["id"])
+      if invoice_item.count == 1
+        invoice_item.first.update_attributes(invoice_item_hash)
+      else
+        InvoiceItem.create!(invoice_item_hash)
+      end
+  end
+end
```

Comment:
@BrianZanti @dionew1

This seems kind of messy, but I spent hours on it, and it's working in dev, so I am going to roll with it. 

I essentially found this online and I understand how it is pulling from the CSV, but I am unfamiliar with the update_attributes action and why exactly that if statement needs to be there. Can either of you explain what is occurring within that if-statement?

Reply:
That's a great question. I'm going to talk to the Invoice Item specifically, but it is what is happening generally with all of them. The line directly before the `if` block is `invoice_item = InvoiceItem.where(id: invoice_item_hash["id"])`.  The variable `invoice_item` is being set to an array of all the invoice items that have a matching id from the current row that you are on.
Now the 'if' block is saying that "if" an invoice item already exists with that id then we are updating.  The else is creating the Invoice Item if it doesn't exist.

In summary, it just a way to handle if there were duplicate ids within the CSV file.